src/AspNet.Security.OAuth.Zalo/ZaloAuthenticationHandler.cs

```diff
@@ -81,9 +81,14 @@ protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OA
             context.Properties.Items.Remove(OAuthConstants.CodeVerifierKey);
         }
 
-        var address = QueryHelpers.AddQueryString(Options.TokenEndpoint, tokenRequestParameters);
+        using var request = new HttpRequestMessage(HttpMethod.Post, Options.TokenEndpoint)
+        {
+            Content = new FormUrlEncodedContent(tokenRequestParameters)
+        };
+
+        Backchannel.DefaultRequestHeaders.Clear();
```

Comment:
Why specifically is this needed?

Reply:
👍🏻 
It's likely not a safe operation anyway: `Backchannel.DefaultRequestHeaders` is not thread-safe so trying to mutate it from a place that can be invoked concurrently is not a good option.

Reply:
I can login with the customize as bellow:
`
var address = QueryHelpers.AddQueryString(Options.TokenEndpoint, tokenRequestParameters);

using var request = new HttpRequestMessage(HttpMethod.Post, address);
request.Headers.Add("secret_key", Options.ClientSecret);
`

But please also take a look at the scope that Zalo allows to access, they not allow to access email anymore @@